zerver/tests/test_reactions.py

```diff
@@ -0,0 +1,145 @@
+        self.assert_json_error(result, 'Emoji does not exist')
+
+    def test_valid_emoji(self):
+        # type: () -> None
+        """
+        Reacting with valid emoji succeeds
+        """
+        sender = 'hamlet@zulip.com'
+        emoji_name = 'my_emoji'
+        emoji_data = {'name': emoji_name, 'url': 'https://example.com/my_emoji'}
+        result = self.client_put('/json/realm/emoji', info=emoji_data,
+                                 **self.api_auth(sender))
+        self.assert_json_success(result)
+        self.assertEqual(200, result.status_code)
+
+        result = self.client_get("/json/realm/emoji", **self.api_auth(sender))
+        content = ujson.loads(result.content)
+        self.assert_json_success(result)
+        print (content)
+        self.assertTrue(emoji_name in content["emoji"])
+
+        result = self.client_post('/api/v1/reactions', {'message_id': 1, 'emoji': emoji_name},
+                                  **self.api_auth(sender))
+        self.assert_json_success(result)
+
+class ReactionMessageIDTest(ZulipTestCase):
+    def test_missing_message_id(self):
+        # type: () -> None
+        """
+        Reacting without a message_id fails
+        """
+        sender = 'hamlet@zulip.com'
+        result = self.client_post('/api/v1/reactions', {'emoji': 'smile'},
+                                  **self.api_auth(sender))
+        self.assert_json_error(result, 'Missing \'message_id\' argument')
+
+    def test_invalid_message_id(self):
+        # type: () -> None
+        """
+        Reacting to an invalid message id fails
+        """
+        sender = 'hamlet@zulip.com'
+        result = self.client_post('/api/v1/reactions', {'message_id': -1, 'emoji': 'smile'},
+                                  **self.api_auth(sender))
+        self.assert_json_error(result, 'Unknown message id')
+
+class ReactionEventTest(ZulipTestCase):
+    def test_event_body(self):
+        # type: () -> None
+        """
+        The reaction event contains the emoji, the message id and the sender
+        """
+        message_sender_email = 'hamlet@zulip.com'
+        self.login(message_sender_email)
+        last_message = self.get_last_message()
+        reaction_sender_email = 'othello@zulip.com'
+        events = [] # type: List[Dict[str, Any]]
+        with tornado_redirected_to_list(events):
+            result = self.client_post('/api/v1/reactions', {'message_id': last_message.id,
+                                                            'emoji': 'smile'},
+                                      **self.api_auth(reaction_sender_email))
+        self.assert_json_success(result)
+        self.assertEqual(len(events), 1)
+
+        event = events[0]['event']
+
+        self.assertEqual(event['sender']['email'], reaction_sender_email)
+        self.assertEqual(event['type'], 'reaction')
+        self.assertEqual(event['emoji_name'], 'smile')
+        self.assertEqual(event['message_id'], last_message.id)
```

Comment:
I'm not sure this test adds much over the one after it?  Seems like you could just keep the second test.